### DISCUSSION TITLE
How to set clipboard from Page Action?

### DISCUSSION THREAD

**Discussion by aurawindsurfing** [filamentphp/filament#5524] [General]
Hi,

I'm wondering if I'm missing something neat that @danharrin already thought about! I know pages are Livewire components so I should be able to emit event from Page Action, push javascript script to the stack and this way set clipboard.

I'm just wondering is there nicer, more neat way of achieving it?

```
    protected function getActions(): array
    {
        return [
            Action::make('Generate readers digests')
                ->icon('heroicon-s-clipboard-list')
                ->action(function (){

                    $this->emit('copyToClipboard');

                Notification::make()
                    ->title('Copied to clipboard')
                    ->success()
                    ->send();
            }),
        ];
    }
```

Thanks for all the suggestions!

**Comment by danharrin** [comment]
Probably `extraAttributes(['x-clipboard.raw' => 'Content here'])`

**Reply by aurawindsurfing** [reply]
Nope, did not work. Besides I would like to do it from within the action callback.

**Reply by danharrin** [reply]
`dispatchBrowserEvent()` is probably best then, and listen for that browser event in JS

**Reply by aurawindsurfing** [reply]
Ok cool. Maybe something to look into in the 3 of actions abilities. I actually realised I was copying to clipboard only to paste it into Trix so just appended it to the Trix field and done!
